coriolis/tests/api/v1/test_replica_schedules.py

```diff
@@ -0,0 +1,400 @@
+        self,
+        mock_validate_schedule,
+        mock_validate_value,
+        mock_validate_expiration_date
+    ):
+        schedule = mock.sentinel.schedule
+        date = mock.sentinel.date
+        mock_body = {
+            "schedule": schedule,
+            "enabled": False,
+            "expiration_date": date,
+            "shutdown_instance": True
+        }
+        expected_result = (
+            mock_validate_schedule.return_value,
+            False,
+            mock_validate_expiration_date.return_value,
+            True
+        )
+
+        result = self.replica_schedules._validate_create_body(mock_body)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+        mock_validate_schedule.assert_called_once_with(schedule)
+        mock_validate_value.assert_called_once()
+        mock_validate_expiration_date.assert_called_once_with(date)
+
+    def test_validate_create_body_no_schedule(
+        self,
+    ):
+        schedule = None
+        mock_body = {"schedule": schedule}
+
+        self.assertRaises(
+            exception.InvalidInput,
+            self.replica_schedules._validate_create_body,
+            mock_body
+        )
+
+    @mock.patch.object(replica_schedules.ReplicaScheduleController,
+                       '_validate_expiration_date')
+    @mock.patch.object(schemas, 'validate_value')
+    @mock.patch.object(replica_schedules.ReplicaScheduleController,
+                       '_validate_schedule')
+    def test_validate_update_body(
+        self,
+        mock_validate_schedule,
+        mock_validate_value,
+        mock_validate_expiration_date
+    ):
+        schedule = mock.sentinel.schedule
+        date = mock.sentinel.date
+        mock_update_body = {
+            "schedule": schedule,
+            "enabled": False,
+            "expiration_date": date,
+            "shutdown_instance": True
+        }
+        expected_result = {
+            "schedule": mock_validate_schedule.return_value,
+            "enabled": False,
+            "expiration_date": mock_validate_expiration_date.return_value,
+            "shutdown_instance": True
+        }
+
+        result = self.replica_schedules._validate_update_body(mock_update_body)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+        mock_validate_schedule.assert_called_once_with(schedule)
+        mock_validate_value.assert_called_once()
+        mock_validate_expiration_date.assert_called_once_with(date)
+
+    @mock.patch.object(schemas, 'validate_value')
+    def test_validate_update_body_none(
+        self,
+        mock_validate_value
+    ):
+        mock_update_body = {}
+        expected_result = {}
+
+        result = self.replica_schedules._validate_update_body(mock_update_body)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+        mock_validate_value.assert_called_once()
```

Comment:
Same as above. Please also mock the other validation methods and assert they're not called